ambry-network/src/main/java/com.github.ambry.network/ConnectionTracker.java

```diff
@@ -56,21 +61,85 @@ boolean mayCreateNewConnection(String host, Port port) {
   }
 
   /**
-   * Start tracking a new connection id associated with the given host and port. Note that this connection will not
-   * be made available for checking out until a {@link #checkInConnection(String)} is called on it.
+   * Configure the connection tracker to keep a specified percentage of connections to this (host, port) ready for use.
+   * @param host the hostname.
+   * @param port the port number.
+   * @param warmUpPercentage percentage of max connections to this (host, port) that should be kept ready for use.
+   */
+  void enableConnectionWarmUp(String host, Port port, int warmUpPercentage) {
```

Comment:
Feel more comfortable to call it `setLowWatermark `, but open to it.

Reply:
Changed the codebase to use `minActiveConnections` instead of `lowWatermark`